lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -2248,13 +2251,19 @@ private void setValueForLabel(TooltipDrawable label, float value) {
     int top = calculateTrackCenter() - (labelPadding + thumbRadius);
     label.setBounds(left, top - label.getIntrinsicHeight(), left + label.getIntrinsicWidth(), top);
 
-    // Calculate the difference between the bounds of this view and the bounds of the root view to
-    // correctly position this view in the overlay layer.
-    Rect rect = new Rect(label.getBounds());
-    DescendantOffsetUtils.offsetDescendantRect(ViewUtils.getContentView(this), this, rect);
-    label.setBounds(rect);
+    if (labelBehavior != LABEL_VISIBLE) {
+      // Calculate the difference between the bounds of this view and the bounds of the root view to
+      // correctly position this view in the overlay layer.
+      Rect rect = new Rect(label.getBounds());
+      DescendantOffsetUtils.offsetDescendantRect(ViewUtils.getContentView(this), this, rect);
+      label.setBounds(rect);
+    }
 
-    ViewUtils.getContentViewOverlay(this).add(label);
+    if (labelBehavior != LABEL_VISIBLE) {
+      ViewUtils.getContentViewOverlay(this).add(label);
+    } else {
+      label.draw(canvas);
```

Comment:
so for this, the problem we have is that is if we don't draw on the view overlay, we can not draw out of the view bounds, unless we set clipChildren to false on the container (which we can't do from this class). Ideally the solution would possibly be to hide the label while the container is being scrolled

Reply:
Yea that's a good observation. I think drawing the label within the view makes more sense in this case. I don't think users would want to draw the label outside of the view if it's persistent, since in that case it would likely be covering up an adjacent view.